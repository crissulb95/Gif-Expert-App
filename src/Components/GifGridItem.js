import React from 'react'
import PropTypes from 'prop-types';

const GifGridItem = ({ img }) => {

    const { title, url } = img;

    return (
        <div className="card animate__animated animate__fadeIn">
            <h3>{ title }</h3>
            <img src={ url } alt={ title } />
        </div>
    )
}

GifGridItem.propTypes = {
    img : PropTypes.object
};

export default GifGridItem;

