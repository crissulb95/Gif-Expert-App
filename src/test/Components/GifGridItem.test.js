import React from 'react';
import { shallow } from 'enzyme';

import GifGridItem from '../../Components/GifGridItem';

describe( 'Pruebas en componente <GifGridItem />', () => {
    const title = 'Un t√≠tulo';
    const url   = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    

    test( 'Debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    
    test('Heading con el titulo', () => {
        const h3 = wrapper.find('h3');
        expect( h3.text().trim() ).toBe( title );
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });


    test('debe de tener clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );
    })

});