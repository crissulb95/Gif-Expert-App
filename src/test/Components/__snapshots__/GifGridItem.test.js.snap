// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <h3 />
  <img />
</div>
`;
